extensions/amp-viewer-integration/0.1/test/test-touch-handler.js

```diff
@@ -114,6 +114,23 @@ describes.fakeWin('TouchHandler', {}, (env) => {
       expect(messages[0].data.data.type).to.equal('touchstart');
     });
 
+    it('should not forward events marked with `shouldViewerCancelPropagation`', () => {
+      const event = fakeTouchEvent('touchmove');
+      event.stopImmediatePropagation = env.sandbox.spy();
+
+      touchHandler.handleEvent_(event);
+      expect(messages).to.have.length(1);
+      expect(messages[0].data.data.type).to.equal('touchmove');
+      expect(event.stopImmediatePropagation).to.not.be.called;
+
+      event.shouldViewerCancelPropagation = true;
```

Comment:
Do you think this test could be separated out a bit to ensure it's per touch event?

Case 1: Never stop propagation.
Case 2: Stop propagation once.
Case 3: Allow propagation once, stop propagation once.
Case 4: Stop propagation once, allow propagation once.